_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder_va.cpp

```diff
@@ -101,29 +101,32 @@ namespace UMC_MPEG2_DECODER
         // iterate through frames submitted to the driver in decoded order
         for (MPEG2DecoderFrame* frm : decode_queue)
         {
-            uint16_t surfCorruption = 0;
-            auto sts = m_packer->SyncTask(frm, &surfCorruption);
-
             frm->CompleteDecoding(); // complete frame even if there are errors
 
-            if (sts < UMC::UMC_OK)
+            if (m_va->IsUseStatusReport())
```

Comment:
I believe just "if (!m_va->IsUseStatusReport()) return;" will look better, less {} depth.

Reply:
@dmitryermilov +1, but `continue` instead of `return`

Reply:
Moreover, I'd rather move frame completion (incl. status reporting) into separate private method, something like
```
void MPEG2DecoderVA::CompleteFrame(MPEG2DecoderFrame* frm)
{
    assert(frm);

    frm->CompleteDecoding();

    if (!m_va->IsUseStatusReport())
        return;
    //...
}

bool MPEG2DecoderVA::QueryFrames(MPEG2DecoderFrame& frame)
{
    //...
    std::for_each(std::begin(decode_queue), std::end(decode_queue),
        [this](MPEG2DecoderFrame* frm) { CompleteFrame(frm); }
    );
    //...
}
```